Modules/Common/AvalancheLikeCrossChainModule.php

```diff
@@ -0,0 +1,308 @@
+<?php declare(strict_types = 1);
+
+/*  Idea (c) 2023 Nikita Zhavoronkov, nikzh@nikzh.com
+ *  Copyright (c) 2023 3xpl developers, 3@3xpl.com, see CONTRIBUTORS.md
+ *  Distributed under the MIT software license, see LICENSE.md  */
+
+/*  This module processes Avalanche cross-chain transfers that involve C-chain on either side.
+ *  Special microservice API by Blockchair is needed (see https://github.com/Blockchair/avax-atomic-unpacker). */
+
+abstract class AvalancheLikeCrossChainModule extends CoreModule
+{
+    use EVMTraits;
+
+    public ?BlockHashFormat $block_hash_format = BlockHashFormat::HexWith0x;
+    public ?AddressFormat $address_format = AddressFormat::HexWith0x;
+    public ?TransactionHashFormat $transaction_hash_format = TransactionHashFormat::HexWith0x;
+    public ?TransactionRenderModel $transaction_render_model = TransactionRenderModel::UTXO;
+    public ?CurrencyFormat $currency_format = CurrencyFormat::AlphaNumeric;
+    public ?CurrencyType $currency_type = CurrencyType::FT;
+    public ?FeeRenderModel $fee_render_model = FeeRenderModel::ExtraBF;
+    public ?PrivacyModel $privacy_model = PrivacyModel::Transparent;
+
+    public ?array $events_table_fields = ['block', 'transaction', 'sort_key', 'time', 'currency', 'address', 'effect', 'extra'];
+    public ?array $events_table_nullable_fields = ['transaction', 'extra'];
+
+    public ?ExtraDataModel $extra_data_model = ExtraDataModel::Type;
+    public ?array $extra_data_details = [
+        'i' => 'Import',
+        'x' => 'Export',
+        'b' => 'Burnt fee',
+    ];
+
+    public ?array $currencies_table_fields = ['id', 'name', 'symbol', 'decimals'];
+    public ?array $currencies_table_nullable_fields = ['symbol'];
+
+    public ?bool $should_return_events = true;
+    public ?bool $should_return_currencies = true;
+    public ?bool $allow_empty_return_events = true;
+    public ?bool $allow_empty_return_currencies = true;
+
+    public ?bool $mempool_implemented = false;
+    public ?bool $forking_implemented = false;
+
+    // EVM-inherited
+    public array $extra_features = [];
+
+    // Avalanche-specific
+    public ?string $unpacker_handle = null;
+    public ?string $asset_info_handle = null;
+    public ?array $main_token_descr = null;
+    public ?array $sister_networks = null;
```

Comment:
let's rename this attribute to subnets please @Lorgansar

Reply:
Renamed to `sister_chains`: in AVAX ecosystem subnet can be comprised of several blockchains, name "chains" should be more reflective of this module's nature.